src/tespy/components/turbomachinery/steam_turbine.py

```diff
@@ -0,0 +1,292 @@
+    to 0,5 bar at the outlet. For example, it is possible to calulate the power
+    output and vapour content at the outlet for a given isentropic efficiency.
+
+    >>> from tespy.components import Sink, Source, Turbine
+    >>> from tespy.connections import Connection
+    >>> from tespy.networks import Network
+    >>> from tespy.tools import ComponentCharacteristics as dc_cc
+    >>> import shutil
+    >>> nw = Network(p_unit='bar', T_unit='C', h_unit='kJ / kg', iterinfo=False)
+    >>> si = Sink('sink')
+    >>> so = Source('source')
+    >>> st = SteamTurbine('steam turbine')
+    >>> st.component()
+    'turbine'
+    >>> inc = Connection(so, 'out1', st, 'in1')
+    >>> outg = Connection(st, 'out1', si, 'in1')
+    >>> nw.add_conns(inc, outg)
+
+    In design conditions the isentropic efficiency is specified.
+    >>> st.set_attr(eta_s=0.9)
+    >>> inc.set_attr(fluid={'water': 1}, m=10, T=250, p=20)
+    >>> outg.set_attr(p=0.1)
+    >>> nw.solve('design')
+    >>> round(st.P.val, 0)
+    -7471296.0
+    >>> round(outg.x.val, 3)
+    0.821
+
+    To capture the effect of liquid drop-out on the isentropic
+    efficiency, the dry turbine efficiency is specified
+    >>> st.set_attr(eta_s=None)
+    >>> st.set_attr(eta_dry_s=0.9, alpha=1.0)
+    >>> nw.solve('design')
+    >>> round(st.P.val, 0)
+    -7009682.0
+    >>> round(outg.x.val, 3)
+    0.840
+    """
+
+    @staticmethod
+    def component():
+        return 'steam turbine'
+
+    def get_parameters(self):
+
+        params = super().get_parameters()
+        params["alpha"] = dc_cp(min_val=0.4, max_val=2.5)
+        params["eta_s_dry"] = dc_cp(min_val=0.0, max_val=1.0)
+        params["eta_s_dry_group"] = dc_gcp(
+            num_eq=1, elements=["alpha", "eta_s_dry"],
+            func=self.eta_s_wet_func,
+            deriv=self.eta_s_wet_deriv,
+            latex=self.eta_s_wet_func_doc)
+
+        return params
+
+    def eta_s_wet_func(self):
+        r"""
+                Equation for given dry isentropic efficiency of a turbine under wet expansion.
+
+                Returns
+                -------
+                residual : float
+                    Residual value of equation.
+
+                    .. math::
+
+                        0 = -\left( h_{out} - h_{in} \right) +
+                        \left( h_{out,s} - h_{in} \right) \cdot \eta_{s,e}
+
+                        \eta_{s,e} = \eta_{s,e}^{dry} \cdot \left( 1 - \alpha \cdot y_m \right)
+
+                        y_m = \frac{\left( 1-x_{in}\right)+ \left( 1-x_{out} \right)}{2}
+                """
+
+        inl = self.inl[0]
+        outl = self.outl[0]
+
+        state = inl.calc_phase()
+        if state == "tp":  # two-phase or saturated vapour
+
+            ym = 1 - (inl.calc_x() + outl.calc_x()) / 2  # average wetness
+            self.eta_s.val = self.eta_dry.val * (1 - self.alpha.val * ym)
+
+            return self.eta_s_func()
```

Comment:
Question: If we start at saturation or in two phase, we make the assumption, that the expansion path will be wet all the way, correct?

Reply:
P. S.: I kind of dislike the fact, that the function here has a side-effect on a parameter of the component by setting the `eta_s.val` of the component. I would actually suggest, `return self.calc_eta_s() - self.eta_s_dry.val * ...`. Here the reformulation of the isentropic efficiency function does not have any benefits because all residuals will be calculated numerically anyways.

Reply:
I'll implement this real quick and push it to your branch

Reply:
done